ledger/src/blockstore.rs

```diff
@@ -866,6 +874,128 @@ impl Blockstore {
         }
     }
 
+    /// Fetches (and populates if needed) the DoubleMerkleMeta for the given block.
+    /// Returns the double_merkle_root
+    ///
+    /// Should only be used on full blocks.
+    pub fn get_or_compute_double_merkle_root(
+        &self,
+        slot: Slot,
+        block_location: BlockLocation,
+    ) -> std::result::Result<Hash, BlockstoreProcessorError> {
+        if let Some(double_merkle_meta) = self
+            .double_merkle_meta_cf
+            .get((slot, block_location))
+            .expect("Blockstore operations must succeed")
+        {
+            return Ok(double_merkle_meta.double_merkle_root);
+        }
+
+        // Compute double merkle - slot must be full at this point
+        let Some(slot_meta) = self
+            .meta_cf
+            .get(slot)
+            .expect("Blockstore operations must succeed")
+        else {
+            return Err(BlockstoreProcessorError::FailedToLoadMeta);
+        };
```

Comment:
Rather than let-else-return error, we can do (here and elsewhere in the function):

```suggestion
        let slot_meta = self
            .meta_cf
            .get(slot)
            .expect("Blockstore operations must succeed")
            .ok_or(BlockstoreProcessorError::FailedToLoadMeta)?;
```